Foundation/FileManager+Win32.swift

```diff
@@ -299,9 +299,58 @@ extension FileManager {
     }
 
     internal func _destinationOfSymbolicLink(atPath path: String) throws -> String {
-        return try _canonicalizedPath(toFileAtPath: path)
+        let handle = path.withCString(encodedAs: UTF16.self) { symlink in
+            CreateFileW(symlink,
+                GENERIC_READ,
+                DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE),
+                nil,
+                DWORD(OPEN_EXISTING),
+                DWORD(FILE_FLAG_OPEN_REPARSE_POINT | FILE_FLAG_BACKUP_SEMANTICS),
+                nil)
+        }
+
+        guard handle != INVALID_HANDLE_VALUE else {
+          throw _NSErrorWithWindowsError(GetLastError(), reading: true)
+        }
+        defer { CloseHandle(handle) }
+
+        // Since REPARSE_DATA_BUFFER ends with an arbitrarily long buffer, we
+        // have to manually get the path buffer out of it since binding it to a
+        // type will truncate the path buffer
+        let pathBufferOffset =
+            MemoryLayout<ULONG>.size //ULONG ReparseTag
+            + MemoryLayout<USHORT>.size //ULONG ReparseDataLength
+            + MemoryLayout<USHORT>.size //ULONG Reserved
+            + MemoryLayout<USHORT>.size //ULONG SubstituteNameOffset
+            + MemoryLayout<USHORT>.size //ULONG SubstituteNameLength
+            + MemoryLayout<USHORT>.size //ULONG PrintNameOffset
+            + MemoryLayout<USHORT>.size //ULONG PrintNameLength
+            + MemoryLayout<ULONG>.size //ULONG Flags
+
+        let buffSize = MemoryLayout<REPARSE_DATA_BUFFER>.size + Int(2 * MAX_PATH) * MemoryLayout<wchar_t>.size
```

Comment:
Why `2 * MAX_PATH`?  Can you use `WCHAR` rather than `wchar_t` please?  `wchar_t` is differently sized on different targets, `WCHAR` is a typedef for `unsigned short`.

Reply:
Microsoft actually defines a `MAXIMUM_REPARSE_DATA_BUFFER_SIZE` constant which we should use here. `MAX_PATH` is irrelevant.